ls1intum/Artemis
# `Programming exercises`: Fix an issue where the checkbox to allow complaints on automatic assessment was incorrectly disabled

### Checklist
#### General



### Motivation and Context
@Strohgelaender noticed that the checkbox was disabled in cases where it should be enabled.

### Description
The condition to disable the checkbox was extended with an additional check whether it's an exam exercise or not.
### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with complaints enabled

Generate a programming exercise and make sure that the checkbox to allow complaints for automatic assessments is enabled if the assessment mode is automatic and a due date is set and disabled in all other cases.

#### Exam Mode Testing



Prerequisites:
- 1 Instructor
- 1 Exam

1. Generate a programming exercise and make sure that the checkbox to allow complaints for automatic assessments is enabled if the assessment mode is automatic and disabled in all other cases.

### Review Progress






#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage





### Screenshots
